eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/EpochSpec.hs

```diff
@@ -304,3 +316,78 @@ depositMovesToTreasuryWhenStakingAddressUnregisters = do
   replicateM_ 5 passEpoch
   expectMissingGovActionId committeeActionId
   expectTreasury $ initialTreasury <> govActionDeposit
+
+eventsSpec ::
+  forall era.
+  ( ConwayEraImp era
+  , InjectRuleEvent "TICK" ConwayEpochEvent era
+  , Event (EraRule "NEWEPOCH" era) ~ ConwayNewEpochEvent era
+  , Event (EraRule "EPOCH" era) ~ ConwayEpochEvent era
+  ) =>
+  SpecWith (ImpTestState era)
+eventsSpec = describe "Events" $ do
+  describe "emits event" $ do
+    it "GovInfoEvent" $ do
+      (drepCred, ccCred, _) <- electBasicCommittee
+      let actionLifetime = 10
+      modifyPParams $ \pp ->
+        pp
+          & ppGovActionLifetimeL .~ EpochInterval actionLifetime
+          & ppDRepVotingThresholdsL .~ def {dvtPPEconomicGroup = 1 %! 2}
+      propDeposit <- getsNES $ nesEsL . curPParamsEpochStateL . ppGovActionDepositL
+      let
+        proposeCostModel = do
+          newTau <- arbitrary
+          submitGovAction @era $
```

Comment:
You could probably use `submitParameterChange` if you wanted, though not sure if it would bring much more conciseness